ddtrace/internal/telemetry/writer.py

```diff
@@ -145,15 +147,14 @@ def add_integration(self, integration_name, auto_enabled):
         :param bool auto_enabled: True if module is enabled in _monkey.PATCH_MODULES
         """
         with self._lock:
-            if not self._enabled:
+            if self._enabled is not None and not self._enabled:
                 return
 
             integration = {
                 "name": integration_name,
                 "version": "",
                 "enabled": True,
                 "auto_enabled": auto_enabled,
-                "compatible": "",
```

Comment:
why is this needed?

Reply:
This is needed because `compatible` is an optional boolean in our schema.
There is currently a bug, I'm fixing https://github.com/DataDog/dd-go/pull/58907 that allows this payload to be accepted by the backend but it's a mistake.

I guess this PR is starting to touch multiple orthogonal things, but I think I'd rather fix everything once than making a bunch of smaller ones. Wdyt?

Reply:
> I guess this PR is starting to touch multiple orthogonal things, but I think I'd rather fix everything once than making a bunch of smaller ones. Wdyt?

Yeah, this is fine. I'll mark the PR as `manual merge` to make sure this doesn't get merged, but can use this to stage all the changes then we can figure out how to break up later!